saleor/core/transactions.py

```diff
@@ -0,0 +1,18 @@
+from contextlib import contextmanager
+
+from django.db import DatabaseError, IntegrityError, transaction
+
+
+@contextmanager
+def transaction_with_commit_on_errors():
+    """Perform transaction and raise an error in any occurred."""
+    error = None
+    with transaction.atomic():
+        try:
+            yield
+        except (DatabaseError, IntegrityError):
```

Comment:
`IntegrityError` is a subclass of `DatabaseError`.

Reply:
Yes, I know but I thought that passing it here explicitly will be a better way

Reply:
To me, it's confusing right now and caused me to double-check if there were maybe different classes with the same names.

Reply:
Okay, I will remove `IntegrityError`